src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtilsTest.java

```diff
@@ -301,6 +301,28 @@ public void shouldUnwrapMultipleParents() {
         assertEquals("maven:2", mavenTemplate.getImage());
     }
 
+    @Test
+    public void shouldCombineInitContainers() {
+        Pod parentPod = new PodBuilder()
+                .withNewMetadata().endMetadata()
+                .withNewSpec()
+                    .withInitContainers(new ContainerBuilder().withName("init-parent").build())
+                .endSpec()
+                .build();
+        Pod childPod = new PodBuilder()
+                .withNewMetadata().endMetadata()
+                .withNewSpec()
+                .withInitContainers(new ContainerBuilder().withName("init-child").build())
+                .endSpec()
+                .build();
+
+        Pod combinedPod = combine(parentPod, childPod);
+        List<Container> initContainers = combinedPod.getSpec().getInitContainers();
+        assertThat(initContainers, hasSize(2));
+        assertThat(initContainers.get(0).getName(), equalTo("init-parent"));
+        assertThat(initContainers.get(1).getName(), equalTo("init-child"));
```

Comment:
Should test the case where a child has a container with the same name as one of the parent containers.